lib/lambda.js

```diff
@@ -2,11 +2,301 @@
+    }
+  }
+  return faas
+}
+
+function setApiGatewayData (trans, event, context, isColdStart) {
+  const faas = getSharedFaasData(context, isColdStart)
+  faas.trigger.request_id = event.requestContext.requestId
+  faas.trigger.type = 'http'
+  trans.setFaas(faas)
+
+  trans.type = 'request'
+
+  // handle structure of both V1 and V2 versions
+  // of Amazon's Gateway API event objects
+  let httpMethod = event.requestContext.httpMethod
+  if (!httpMethod) {
+    httpMethod = event.requestContext.http && event.requestContext.http.method
+  }
+  let path = event.requestContext.resourcePath
+  if (!path) {
+    path = event.requestContext.http && event.requestContext.http.path
+    if (path) {
+      // remove stage
+      path = path.replace('/' + event.requestContext.stage, '')
+    }
+  }
+
+  const stage = event.requestContext.stage
+  const apiId = event.requestContext.apiId
+  trans.name = httpMethod + ' ' + context.functionName
+  const version = (event && event.version) ? event.version : '1.0'
+  const serviceContext = {
+    origin: {
+      name: `${httpMethod} ${path}/${stage}`,
+      id: apiId,
+      version: version
+    }
+  }
+  trans.setServiceContext(serviceContext)
+
+  const cloudContext = {
+    origin: {
+      provider: 'aws',
+      service: {
+        name: 'api gateway'
+      },
+      account: {
+        id: event && event.requestContext.accountId
+      }
+    }
+  }
+  trans.setCloudContext(cloudContext)
+}
+
+function isApiGatewayEvent (event) {
+  return !!(event && event.requestContext && event.requestContext.requestId)
+}
+
+function setSqsData (agent, trans, event, context, isColdStart) {
+  const record = event && event.Records && event.Records[0]
+  const eventSourceARN = record.eventSourceARN ? record.eventSourceARN : ''
+
+  const faas = getSharedFaasData(context, isColdStart)
+  faas.trigger.request_id = record.messageId
+  faas.trigger.type = 'pubsub'
+  trans.setFaas(faas)
+
+  const arnParts = eventSourceARN.split(':')
+  const queueName = arnParts[5]
+  const accountId = arnParts[4]
+
+  trans.name = `RECEIVE ${queueName}`
+  trans.type = 'messaging'
+
+  const serviceContext = {
+    origin: {
+      name: queueName,
+      id: eventSourceARN
+    }
+  }
+  trans.setServiceContext(serviceContext)
+
+  const cloudContext = {
+    origin: {
+      provider: 'aws',
+      region: record.awsRegion,
+      service: {
+        name: 'sqs'
+      },
+      account: {
+        id: accountId
+      }
+    }
+  }
+  trans.setCloudContext(cloudContext)
+
+  let age
+  if (record.attributes && record.attributes.SentTimestamp) {
+    age = Date.now() - record.attributes.SentTimestamp
+  }
+  const messageContext = {
+    queue: {
+      name: record.eventSourceARN,
+    },
+    age: {
+      ms: age,
+    },
+    body: undefined,
+    headers: undefined
+  }
+
+  if(agent._conf.captureHeaders) {
+    messageContext.headers = Object.assign({}, record.messageAttributes ? record.messageAttributes : {})
```

Comment:
nit:

```suggestion
    messageContext.headers = Object.assign({}, record.messageAttributes)
```

A falsy value for an argument to Object.assign is skipped, so no need for the extra check.